models/classes/class.Variable.php

```diff
@@ -18,182 +21,235 @@
+    public function getCardinality(): ?string
     {
         return $this->cardinality;
     }
 
-    /**
-     * @author  "Patrick Plichart, <patrick@taotesting.com>"
-     * @param string $baseType
-     */
-    public function setBaseType($baseType)
+    public function setBaseType(string $baseType): self
     {
         $this->baseType = $baseType;
+
+        return $this;
     }
 
-    /**
-     * @author  "Patrick Plichart, <patrick@taotesting.com>"
-     * @return string $baseType
-     */
-    public function getBaseType()
+    public function getBaseType(): ?string
     {
         return $this->baseType;
     }
 
-    /**
-     * Set the epoch of the variable
-     * @param $epoch string
-     */
-    public function setEpoch($epoch)
+    public function setEpoch(string $epoch): self
     {
-         $this->epoch = $epoch;
+        $this->epoch = $epoch;
+
+        return $this;
     }
 
-    /**
-     * Get the epoch of a variable
-     * @return string
-     */
-    public function getEpoch()
+    public function getEpoch(): ?string
     {
-         return $this->epoch;
+        return $this->epoch;
     }
 
-    /**
-     * Returns the time of the creation
-     * @return float|null
-     */
-    public function getCreationTime()
+    public function getCreationTime(): ?float
     {
         if (!isset($this->epoch)) {
             return null;
         }
-        list($usec, $sec) = explode(" ", $this->epoch);
-        return floatval((float) $usec + (float) $sec);
+        [$usec, $sec] = explode(' ', $this->epoch);
+
+        return ((float)$usec + (float)$sec);
     }
 
-    /**
-     * Allow to know if the epoch is set or not
-     * @return bool
-     */
-    public function isSetEpoch()
+    public function isSetEpoch(): bool
     {
         return (isset($this->epoch));
     }
 
-    /**
-     * Check if variable is of multiple type.
-     *
-     * @return bool
-     */
-    public function isMultiple()
+    public function isMultiple(): bool
     {
-        return in_array($this->cardinality, [self::CARDINALITY_MULTIPLE, self::CARDINALITY_ORDERED]);
+        return in_array($this->cardinality, [self::CARDINALITY_MULTIPLE, self::CARDINALITY_ORDERED], true);
     }
 
     /**
      * get the value of the variable
+     *
      * @return mixed
      */
     abstract public function getValue();
 
     /**
-     * set the value of the variable
+     * Set the value of the variable
+     *
      * @param $value mixed
      */
     abstract public function setValue($value);
 
     /**
-     * Get the json representation of the variable
-     * @return string
+     * @deprecated Use jsonSerialize method instead
+     */
+    public function toJson(): string
+    {
+        return json_encode($this->jsonSerialize());
+    }
+
+    /**
+     * {@inheritdoc}
      */
-    public function toJson()
+    public function jsonSerialize()
     {
-        return json_encode((array)$this);
+        return [
```

Comment:
you could use return `get_object_vars($this)` probably since you seems to return all properties